src/gluonts/model/deepvar/_network.py

```diff
@@ -0,0 +1,822 @@
+        lags_seq.sort()
+
+        self.lags_seq = lags_seq
+
+        self.distr_output = distr_output
+
+        self.target_dim = target_dim
+
+        with self.name_scope():
+            self.proj_dist_args = distr_output.get_args_proj()
+
+            residual = True
+
+            self.rnn = make_rnn_cell(
+                cell_type=cell_type,
+                num_cells=num_cells,
+                num_layers=num_layers,
+                residual=residual,
+                dropout_rate=dropout_rate,
+            )
+
+            self.embed_dim = 1
+            self.embed = mx.gluon.nn.Embedding(
+                input_dim=self.target_dim, output_dim=self.embed_dim
+            )
+
+            if scaling:
+                self.scaler = MeanScaler(keepdims=True)
+            else:
+                self.scaler = NOPScaler(keepdims=True)
+
+    @staticmethod
+    def get_lagged_subsequences(
+        F,
+        sequence: Tensor,
+        sequence_length: int,
+        indices: List[int],
+        subsequences_length: int = 1,
+    ) -> Tensor:
+        """
+        Returns lagged subsequences of a given sequence.
+        Parameters
+        ----------
+        sequence
+            the sequence from which lagged subsequences should be extracted.
+            Shape: (N, T, C).
+        sequence_length
+            length of sequence in the T (time) dimension (axis = 1).
+        indices
+            list of lag indices to be used.
+        subsequences_length
+            length of the subsequences to be extracted.
+        Returns
+        --------
+        lagged : Tensor
+            a tensor of shape (N, S, C, I),
+            where S = subsequences_length and I = len(indices),
+            containing lagged subsequences.
+            Specifically, lagged[i, :, j, k] = sequence[i, -indices[k]-S+j, :].
+        """
+        # we must have: history_length + begin_index >= 0
+        # that is: history_length - lag_index - sequence_length >= 0
+        # hence the following assert
+        assert max(indices) + subsequences_length <= sequence_length, (
+            f"lags cannot go further than history length, found lag "
+            f"{max(indices)} while history length is only {sequence_length}"
+        )
+        assert all(lag_index >= 0 for lag_index in indices)
+
+        lagged_values = []
+        for lag_index in indices:
+            begin_index = -lag_index - subsequences_length
+            end_index = -lag_index if lag_index > 0 else None
+            lagged_values.append(
+                F.slice_axis(
+                    sequence, axis=1, begin=begin_index, end=end_index
+                ).expand_dims(axis=1)
+            )
+        return F.concat(
+            *lagged_values, num_args=len(indices), dim=1
+        ).transpose(axes=(0, 2, 3, 1))
+
+    def unroll(
+        self,
+        F,
+        lags: Tensor,
+        scale: Tensor,
+        time_feat: Tensor,
+        target_dimensions: Tensor,
```

Comment:
It's kind of confusing that target_dim and target_dimension refer to two fairly different things.

Reply:
Good point. I'll try to make that clearer. I think the confusion stems mostly from not being really relevant to this model - maybe I'll factor this over to the GP model once, which I will work on next.

Reply:
I refactored a bit and put some description to the estimator constructor. Hope that is clearer.